controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,359 @@
+KubernetesClusterID=%q
+Zone=%q
+`,
+		infraStatus.VPC.ID, subnetID, cp.Namespace, cp.Namespace, zone)
+	values := map[string]interface{}{
+		"cloudProviderConfig": cloudProviderConfig,
+	}
+
+	// Apply config chart
+	a.logger.Info("Applying configuration chart", "controlplane", objectName(cp), "chart", configChart.Name, "values", values)
+	err = configChart.Apply(ctx, a.gcs, a.chartApplier, cp.Namespace, cluster.Shoot, nil, nil, values)
+	if err != nil {
+		return errors.Wrapf(err, "could not apply configuration chart for controlplane '%s'", objectName(cp))
+	}
+
+	// Get cloudprovider secret and configmap from cluster
+	cpSecret, err := a.getSecret(cp.Namespace, common.CloudProviderSecretName)
+	if err != nil {
+		return err
+	}
+	senConfigMap, err := a.getConfigMap(cp.Namespace, common.CloudProviderConfigName)
+	if err != nil {
+		return err
+	}
+
+	// Compute checksums
+	csSecrets := util.MergeSecretMaps(deployedSecrets, map[string]*corev1.Secret{
+		common.CloudProviderSecretName: cpSecret,
+	})
+	csConfigMaps := map[string]*corev1.ConfigMap{
+		common.CloudProviderConfigName: senConfigMap,
+	}
+	checksums := util.ComputeChecksums(csSecrets, csConfigMaps)
+
+	// Collect additional CCM chart values
+	values = make(map[string]interface{})
+	if cpConfig.CloudControllerManager != nil {
+		values["featureGates"] = cpConfig.CloudControllerManager.FeatureGates
+	}
+
+	// Apply CCM chart
+	a.logger.Info("Applying CCM chart", "controlplane", objectName(cp), "chart", ccmChart.Name, "values", values)
+	err = ccmChart.Apply(ctx, a.gcs, a.chartApplier, cp.Namespace, cluster.Shoot, imagevector.ImageVector, checksums, values)
+	if err != nil {
+		return errors.Wrapf(err, "could not apply CCM chart for controlplane '%s'", objectName(cp))
+	}
+
+	return nil
+}
+
+// Delete reconciles the given controlplane and cluster, deleting the additional Shoot
+// control plane components as needed.
+func (a *actuator) Delete(
+	ctx context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) error {
+	// Delete CCM objects
+	a.logger.Info("Deleting CCM objects", "controlplane", objectName(cp))
+	if err := ccmChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete CCM objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete config objects
+	a.logger.Info("Deleting configuration objects", "controlplane", objectName(cp))
+	if err := configChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete configuration objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete secrets
+	a.logger.Info("Deleting secrets", "controlplane", objectName(cp))
+	if err := controlPlaneSecrets.Delete(a.cs, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete secrets for controlplane '%s'", objectName(cp))
+	}
+
+	return nil
+}
+
+func (a *actuator) getSecret(namespace, name string) (*corev1.Secret, error) {
```

Comment:
Hm, why do we need a wrapper function here? Also, I'd rather prefer to use the controller-runtime client directly instead of the clientset.

Reply:
Refactored as suggested and removed the wrapper functions.